enterprise/dev/ci/internal/ci/operations.go

```diff
@@ -261,30 +271,48 @@ func addBrandedTests(pipeline *bk.Pipeline) {
 		bk.Cmd("dev/ci/codecov.sh -c -F typescript -F unit"))
 }
 
-// This is a bandage solution to speed up the go tests by running the slowest ones
-// concurrently. As a results, the PR time affecting only Go code is divided by two.
-var slowGoTestPackages = []string{
-	"github.com/sourcegraph/sourcegraph/enterprise/internal/codeintel/stores/dbstore",   // 224s
-	"github.com/sourcegraph/sourcegraph/enterprise/internal/codeintel/stores/lsifstore", // 122s
-	"github.com/sourcegraph/sourcegraph/enterprise/internal/insights",                   // 82+162s
-	"github.com/sourcegraph/sourcegraph/internal/database",                              // 253s
-	"github.com/sourcegraph/sourcegraph/internal/repos",                                 // 106s
-	"github.com/sourcegraph/sourcegraph/enterprise/internal/batches",                    // 52 + 60
-	"github.com/sourcegraph/sourcegraph/cmd/frontend",                                   // 100s
-}
-
 // Adds the Go test step.
 func addGoTests(pipeline *bk.Pipeline) {
-	pipeline.AddStep(":go: Test",
-		bk.Cmd("./dev/ci/go-test.sh exclude "+strings.Join(slowGoTestPackages, " ")),
-		bk.Cmd("dev/ci/codecov.sh -c -F go"))
+	buildGoTests(func(description, testSuffix string) {
+		pipeline.AddStep(
+			fmt.Sprintf(":go: Test (%s)", description),
+			bk.Cmd("./dev/ci/go-test.sh "+testSuffix),
+			bk.Cmd("./dev/ci/codecov.sh -c -F go"),
+		)
+	})
+}
+
+// Adds the Go backcompat test step.
+func addGoTestsBackcompat(minimumUpgradeableVersion string) func(pipeline *bk.Pipeline) {
+	return func(pipeline *bk.Pipeline) {
+		buildGoTests(func(description, testSuffix string) {
+			pipeline.AddStep(
+				// TODO - set minimum upgradeable version
+				fmt.Sprintf(":go::postgres: Backcompat test (%s)", description),
+				bk.Env("MINIMUM_UPGRADEABLE_VERSION", minimumUpgradeableVersion),
+				bk.Cmd("./dev/ci/go-test-backcompat.sh "+testSuffix),
+			)
+		})
+	}
+}
+
+func buildGoTests(f func(description, testSuffix string)) {
```

Comment:
```suggestion
// buildGoTests ... TODO
func buildGoTests(f func(description, testSuffix string)) {
```

I'm not actually sure how to best describe this - "runs the given callback over all Go packages"?

Reply:
I'll figure something out soon 😉